service/history/queueProcessor.go

```diff
@@ -92,6 +94,7 @@ func newQueueProcessorBase(shard ShardContext, options *QueueProcessorOptions, p
 		metricsClient:           shard.GetMetricsClient(),
 		logger:                  logger,
 		ackMgr:                  queueAckMgr,
+		lastPollTime:            time.Time{},
```

Comment:
Let's keep this within pump rather than exposing it as a field.  
You can return a boolean from processBatch to indicate if it actually polled or not.